packages/expo-camera/ios/ExpoCameraUtils.swift

```diff
@@ -0,0 +1,178 @@
+      return .portrait
+    case .landscapeLeft:
+      return .landscapeLeft
+    case .landscapeRight:
+      return .landscapeRight
+    case .portraitUpsideDown:
+      return .portraitUpsideDown
+    default:
+      return .portrait
+    }
+  }
+
+  static func videoOrientation(for deviceOrientaion: UIDeviceOrientation) -> AVCaptureVideoOrientation {
+    switch deviceOrientaion {
+    case .portrait:
+      return .portrait
+    case .portraitUpsideDown:
+      return .portraitUpsideDown
+    case .landscapeLeft:
+      return .landscapeLeft
+    case .landscapeRight:
+      return .landscapeRight
+    default:
+      return .portrait
+    }
+  }
+
+  static func export(orientation: UIImage.Orientation) -> Int {
+    switch orientation {
+    case .left:
+      return 90
+    case .right:
+      return -90
+    case .down:
+      return 180
+    default:
+      return 0
+    }
+  }
+
+  static func exportImage(orientation: UIImage.Orientation) -> Int {
+    switch orientation {
+    case .left:
+      return 90
+    case .right:
+      return -90
+    case .down:
+      return 180
+    default:
+      return 0
+    }
+  }
+
+  static func generatePhoto(of size: CGSize) -> UIImage {
+    let rect = CGRect(x: 0, y: 0, width: size.width, height: size.height)
+    let renderer = UIGraphicsImageRenderer(size: size)
+
+    return renderer.image { ctx in
+      UIColor.black.setFill()
+      ctx.fill(rect)
+      let currentDate = Date()
+      let dateFormatter = DateFormatter()
+      dateFormatter.dateFormat = "dd.MM.YY HH:mm:ss"
+      let text = dateFormatter.string(from: currentDate)
+      text.draw(
+        with: CGRect(
+          x: size.width * 0.1,
+          y: size.height * 0.9,
+          width: size.width,
+          height: size.height
+        ),
+        attributes: [.font: UIFont.systemFont(ofSize: 18), .foregroundColor: UIColor.orange],
+        context: nil)
+    }
+  }
+
+  static func crop(image: UIImage, to rect: CGRect) -> UIImage {
+    let cgImage = image.cgImage
+    guard let croppedCgImage = cgImage?.cropping(to: rect) else {
+      return image
+    }
+    return UIImage(cgImage: croppedCgImage, scale: image.scale, orientation: image.imageOrientation)
+  }
+
+  static func write(data: Data, to path: String) -> String? {
+    let url = URL(fileURLWithPath: path)
+    do {
+      try data.write(to: url, options: .atomic)
+      return url.absoluteString
+    } catch {
+      return nil
+    }
+  }
+
+  static func data(from image: UIImage, with metadata: [String: Any], quality: Float) -> Data? {
+    guard let sourceCGImageRef = image.cgImage,
+    let sourceData = image.jpegData(compressionQuality: 1.0) as CFData?,
+    let sourceCGImageSourceRef = CGImageSourceCreateWithData(sourceData, nil),
+    let sourceMetadata = CGImageSourceCopyPropertiesAtIndex(sourceCGImageSourceRef, 0, nil) as? NSDictionary else {
+      return nil
+    }
+
+    let updatedMetadata = NSMutableDictionary(dictionary: sourceMetadata)
```

Comment:
Do we still need to use the Objective-C classes for dictionaries?

Reply:
I had switched to swift dictionaries but merging the exif data wasn't working properly. Reverted until I figure it out why 😅